docs/src/main/asciidoc/all-builditems.adoc

```diff
@@ -11,6 +11,8 @@ include::_attributes.adoc[]
 :categories: writing-extensions
 :summary: Explore all the BuildItems you can consume/produce in your extensions.
 
-Here you can find a list of Build Items and the extension that provides them:
+Here you can find a list of Build Items and the extension that provides them.
+
+icon:lock[title=Non-instantiatable Build Item] Build item can't be instantiated directly, but can be extended/inherited from
```

Comment:
I am unsure if a 🔒 is the best icon to represent this TBH. Maybe use 🏗️ or similar?

Reply:
ChatGPT doesn't help much either: 

![image](https://github.com/user-attachments/assets/f1832c4d-46e6-46c7-a336-a9bb5431ac7c)

Reply:
I went with the lock for a few reasons:
1. I didn't know of any other one that would work OOTB in asciidoc :)
2. Locked is kinda like abstract. It's locked, meaning can't be changed.

I can change to whatever

Reply:
Similar results with Claude...

![image](https://github.com/user-attachments/assets/764e5d5c-512b-4d83-ad8e-e32e8d37a27d)

Reply:
we're also constrained to the font-awesome library

Reply:
I changed to a building icon
![image](https://github.com/user-attachments/assets/f9969a9d-6228-4511-8c7e-55e05c90ec11)